Website rework

Mstiekema opened the pull request:
I've added a lot of features to the website the past few weeks and I didn't want to work on the looks, because I thought that the features were more important. At the moment, I think that a whole lot of features have been added, but it doesn't look that good. With this PR I'm going to change that. Below you can find the things that I want to work on. Will add more things while working on the rework.

Reworks:
- [x] Collapseable menu
- [x] Rework logo or remove it
- [x] Make the website less plain
- [x] Buttons shouldn't look like normal HTML buttons
- [x] Redesign login menu

New features
- [x] Ability to have a dark or light theme
- [ ] Maybe some localization? 
- [x] Display profile information when you click on your acc name

Mstiekema committed:
Added a favicon

Mstiekema committed:
Improved searchbar and reworked menu

Menu will be improved even more next commit

Mstiekema committed:
Improved menu

Tried to update the songlist history as well

inooid commented on static/css/yucibot.css:
```diff
@@ -124,6 +125,8 @@
 }
 
 #searchBar {
```
I would suggest to minimize the usage of `#id` selectors. The rules say that all ids on html elements should be unique. This means, that you can never re-use this CSS.

You don't have to change it, but keep it in mind 😉

inooid commented on static/css/yucibot.css:
```diff
@@ -208,47 +214,49 @@
-  text-align: center;
+  padding-left: 10px;
+  padding-right: 10px;
+  /*text-align: center;*/
```
Should we remove this comment?

inooid commented on static/html/adminmenu.htm:
```diff
@@ -1,15 +1,15 @@
-  <p><a href="/commands">Commands</a></p>
-  <p><a href="/stats">Stats</a></p>
-  <p><a href="/history/">Songlist History</a></p>
+  <p><a href="/">• Home</a></p>
```
You should probably do this in CSS. You can do this using the `:before` pseudo elements.

The HTML:
```html
<aside class="menu">
  <h3 class="menu_title">Menu</h3>
  <p class="menu_item">
    <a href="/">Home</a>
  </p>
</aside>
```

```css
.menu_item:before {
  content: '• ';
}
```

This will add the content `'• '` before the `.menu_item` element. This will be better for screen readers and Google. You might want to read about semantic html5. A `<p>` element is theoretically not for menu items, they are for paragraphs of text.

The better semantic navigation would look like this:

```html
<aside class="sidebar">
  <section class="menu">
    <h3 class="menu_title">Menu</h3>
    <nav>
      <ul>
        <li><a href="bla">Home</a></li>
      </ul>
    </nav>
  </section>
</aside>
```

Mstiekema replied:
I've looked into the semantic menu, but it kinda ruins the spacing for me. For next commit I've added the menu_item class. Appreciate the help! 😄

inooid replied:
If the spacing is bad, you might want to use something like `reset.css` and this:
 
```css
/* place this at the start of your first css file. */
*, *:before, *:after {
  box-sizing: border-box;
} 
```

Check out `box-sizing` on Google, it's really great!

inooid commented on templates/about.html:
```diff
@@ -26,4 +25,4 @@
 <script type="text/javascript" src="../js/yucisite.js"></script>
-</body>
\ No newline at end of file
+</body>
```
Don't forget to add an extra line-break at the end of the file. Otherwise `git` will think that the last line has changed every time you add/remove lines. If you're using sublime, you can do this automatically on-save.

Open the sublime text settings and add this line: `"ensure_newline_at_eof_on_save": true,`

inooid reviewed (commented):
Couple of small things 😉

inooid commented:
Great work 🎉

Mstiekema committed:
Fixed the player on home page

Mstiekema committed:
Improved buttons

Mstiekema committed:
Improved module list

Mstiekema committed:
Changed h1 to h2

Because it looks way better

inooid commented:
To make the design look a bit better, it might be useful to hang onto a UI library that can help you out. Instead of making everything yourself, it might be better to make use of some library. This way you can learn from their source code, while still building a great product.

Some of the UI libraries off of the top of my head:

- **Bootstrap**: http://getbootstrap.com/
- **Semantic UI**: http://semantic-ui.com/
- **PureCSS**: https://purecss.io/
- **Foundation**: http://foundation.zurb.com/
- **BassCSS**: http://basscss.com/

There's probably plenty more, but this might help you out a little. This also brings more consistency to your design and gives you easy building blocks to build something with 👍 

Think about it. 😁

inooid commented:
I am obviously not saying you **have** to, but it could help you out. It helps you structurize, while not having to deal with specific UI things.

Mstiekema committed:
Improved command page

The more details page will be for point commands in the future. It could
also have some stats for roulette, slot, etc.

Mstiekema commented:
@inooid I've looked into UI libraries, but I want to try to stick to my own conventions for now. I'll maybe use one in the future, but for now I'd like to keep it like this.

Mstiekema committed:
Added a shadow to the player

Mstiekema committed:
Changed menu and updated website colors

I'm actually liking the look of the website right now, took me a while,
but we're finally getting there.

Mstiekema committed:
Fixed the commands details

It didn't work for non everyone commands yet

Mstiekema committed:
Minor changes

Mstiekema committed:
Improved profile and log page

- Easy to add new things to profile page for in the future
- Improved log page
- Search in logs and select logs by date

Mstiekema committed:
Added a light theme

You can easlily switch between the two by clicking on Light or Dark :+1:

inooid commented on install.js:
```diff
@@ -8,6 +8,7 @@ connection.query(
 	'points INT,' +
 	'num_lines INT,' +
 	'level INT,' +
+	'pf VARCHAR(200),' +
```
What is `pf`? 🤔

Mstiekema replied:
With the latest commits I improved the profile page and login menu. One of the new things is using people their profile pictures, `pf` is for storing those. When their user page is loaded, the website fetches the profile picture and inserts or updates it into the user table.

inooid replied:
Why not rename it to something more clear then though 🤔  Like `profile_pic_url`, `profile_pic_path` or just `profile_pic`. Also, can you guarantee that the url is never longer than 200 characters? 🤔

Mstiekema replied:
I just left it there without thinking about it, but yea it should be changed, so I'll update it in the next commit. 
I also think that we can guarantee that it is < 200, because the max length of a Twitch username is 25 chars and the random generated string isn't that long either. I think the max length is 115, but I set it to 200 just to be sure.

inooid commented on static/js/yucisite.js:
```diff
@@ -25,3 +25,12 @@ $("#menuIconOpen").click(function() {
 });
+
+$("#name").click(function() {
+  var classes = $(".profile").attr('class')
```
You can also use `toggleClass` ([read more here](http://api.jquery.com/toggleclass/)).

```javascript
$("#name").click(function() {
  $('.profile').toggleClass('hidden');
});
```

inooid commented on website.js:
```diff
@@ -184,6 +190,9 @@ app.get('/user/:id', function(req, res) {
         }
       }
       request(info, function (error, response, body) {
+        var prePic = JSON.parse(body).logo
```
You should cache the `JSON.parse` call, because it's quite an expensive task.

```javascript
var json = JSON.parse(body); // cache the output of the `JSON.parse` call
var prePic = json.logo;
var name = json.name;

// re-use here
```

inooid reviewed (commented):
Some minor things 😁

Mstiekema committed:
Small improvements

Mstiekema committed:
Improved admin logs

Also changed the naming of the profile picture to be more clear about
its use

Mstiekema committed:
Updated menu

You can click the entire div not instead of only clicking on the text.
Feels way better.

Mstiekema committed:
Merge branch 'master' into website-rework

Mstiekema committed:
Improved poll pages

Also using Roboto as font now, looks way better :+1:

Mstiekema committed:
Updated poll pages

More improvements \o/

Mstiekema committed:
Minor improvements

Mstiekema commented:
I think the website is finished. If spot any bugs or security risks, leave an issue. Thanks for helping me with this update @inooid
